pkg/target/clusters.go

```diff
@@ -72,10 +63,39 @@ func NewClusters(mainCfg conf.Main, cfg conf.Clusters, lg *zap.Logger, ms *metri
 			return cls, fmt.Errorf("incorrect cluster type %s for cluster %s",
 				cl.Type, cl.Name)
 		}
-		cl.Type = cc.Type
 
 		cls[cl.Name] = &cl
 	}
 
-	return cls, nil
+	for _, cl := range cls {
+		go cl.keepAvailableHostsUpdated()
```

Comment:
Why is this executed for all types of clusters? I would expect this to only be executed for the LB clusters. Please correct me if I got something wrong.

Reply:
Good question. I considered the alternative option as well, however, went ahead with this for reasons.
Hosts don't have knowledge of cluster. This way hosts communicate on a channel to the cluster and depending upon cluster type, the cluster can act on them. Currently, only LB acts on them. Alternatively, we would inject cluster into hosts, and try to update cluster based on cluster type. I went ahead with first option.

Reply:
I see that inside keepAvailableHostsUpdated() we are checking for the cluster type.